### DISCUSSION TITLE
How can I listen to a CC value and display it on a TM1637 display?

### DISCUSSION THREAD

**Discussion by justtom01** [tttapa/Control-Surface#679] [Q&A] [answered]
Hello!

I am trying to display the value of a CC on a TM1637 seven segment style display. I am able to get the raw value from the pin and display it using tm.display(value) - however I would like to be able to access the value that control surface is producing, because it already nice and smooth. I can smooth the numbers coming directly from a potentiometer, but I feel that this is unnecessary code because control surface is doing this job already.

How can I capture or listen to the 0-127 value of a CC, and display it using tm.display() ?

Apologies if this is a very newbie question, I have just started my journey.

Kind thanks!

**Comment by tttapa** [comment, marked as answer]
Many of the classes in the library will have some kind of `getValue()` member function that allow you to get the internal value. For example: [`CCPotentiometer::getValue()`](https://tttapa.github.io/Control-Surface-doc/Doxygen/db/d32/classCCPotentiometer.html#a401ab4ab293a05ef4b9d33c8b55472ca).

```cpp
#include <Control_Surface.h> // Include the Control Surface library
USBMIDI_Interface midi;

CCPotentiometer potentiometer { A0, MIDI_CC::Channel_Volume };
 
void setup() {
  Serial.begin(115200);
  Control_Surface.begin();
}
 
void loop() {
  Control_Surface.loop();
  Serial.println(potentiometer.getValue());
}
```

**Reply by justtom01** [reply]
Thankyou very much tttapa, this is exactly what I was looking for.
